src/app/rpmostree-clientlib.cxx

```diff
@@ -1682,7 +1684,31 @@ append_docs_to_str (GString          *str,
     }
 }
 
-/* Check whether sd_unit contains pid and return the boolean in sd_unit_contains_pid */
+/* Check whether sd_unit's `ActiveState` is in `states` and return the boolean
+ * in `sd_unit_is_active`. */
+static gboolean
+check_sd_unit_state (char            *sd_unit,
+                     const char     **states,
+                     gboolean        *states_contains_state,
+                     GDBusConnection *connection,
+                     GCancellable    *cancellable,
+                     GError         **error)
+{
+  g_autoptr(GVariant) active_state = NULL;
+  if (!get_sd_unit_property (connection, "LoadUnit", g_variant_new ("(s)", sd_unit),
+                             "ActiveState", &active_state, cancellable, error))
+    return FALSE;
+  else if (active_state)
+    {
+      const char *active_state_str = g_variant_get_string (active_state, NULL);
+      *states_contains_state = (active_state_str && g_strv_contains (states, active_state_str));
+      return TRUE;
+    }
+  else
+    return FALSE;
```

Comment:
This is returning `FALSE` even if no error actually happened (so `error` is still NULL). Missing a `glnx_throw` here?

Reply:
It looks like `get_sd_unit_property` already handles the missing property case and converts it to an error, so I guess we could instead reflow this function as:

```c
g_autoptr(GVariant) active_state = NULL;
if (!get_sd_unit_property (...))
  return FALSE;
g_assert (active_state); /* checked in get_sd_unit_property */
/* get string & and g_strv_contains() here */
return TRUE;
```

?

Reply:
Ah right. It's in times like this when I start to appreciate Rust a bit more :) Thanks!